psg/src/main.c

```diff
@@ -0,0 +1,78 @@
+// md-toolchain example main.c
+// Michael Moffitt 2018
+//
+// This main shows a simple "hello world" demo.
+
+// megadrive.h is an umbrella for all headers in src/md. Specific modules like
+// md/vdp.h do not need to be individually included. However, utility funcitons
+// are not included, as they are not core support functions.
+#include "md/megadrive.h"
+
+#include "util/text.h"
+#include "util/plane.h"
+
+struct Chord
+{
+	unsigned short period[3];
+	unsigned short duration;
+};
+
+// The base frequency for the PSG squarewaves is
+// 39375000/(11*32) = 111860.8 Hz
+// Thus the following hold:
+// period_value = 111860.8/frequency
+// frequency = 111860.8/period_value
+
+static const struct Chord saygah[] =
+{
+	// E flat, B flat, G
+	111860.8/155.5635, 111860.8/233.0819, 111860.8/391.9954, 30,
+	// C, G, E
+	111860.8/130.8128, 111860.8/195.9977, 111860.8/329.6276, 60,
+	0, 0, 0, 0
+};	
+
+void main(void)
+{
+	const struct Chord *playpos = saygah;
+
+	// Get the megadrive ready to go! (See md/megadrive.h)
+	megadrive_init();
+
+	// Set up text graphics at VRAM address 0x400 palette 0
+	text_init(0x400, 0);
+
+	// Print a simple message in the center of plane A
+	text_puts(VDP_PLANE_A, 15, 11, "PSG Chords");
+
+	// Play a sequence of chords
+	for (const struct Chord *playpos = saygah;
+	     playpos->duration > 0;
+	     ++playpos)
+	{
+		// Set pitch
+		for (unsigned int ch = 0; ch < 3; ++ch) {
```

Comment:
Within a single project (e.g. the scope of one sample) I'm okay with the code style diverging from my own, since this example is yours, and not mine. Here, my only complaint is that it'd be best for it to be internally consistent.

In this case, the opening brace is on the same line as the `for` statement, while the loop below has it on the next line instead.

Between the two, I'd prefer if you picked your favorite and used it for the file. I lean towards the latter, and have thus far in the framework code.

Reply:
Oops, my bad. I tried to follow your style but missed these.